pkg/manager/member/tiflash_member_manager.go

```diff
@@ -212,6 +212,15 @@ func (m *tiflashMemberManager) syncStatefulSet(tc *v1alpha1.TidbCluster) error {
 	}
 	setNotExist := errors.IsNotFound(err)
 
+	// if TiFlash is scale from 0 and with previous StatefulSet, we delete the previous StatefulSet first
+	// to avoid some fileds (e.g storage request) reused and cause unexpected behavior (e.g scale down).
+	if oldSetTmp != nil && *oldSetTmp.Spec.Replicas == 0 && tc.Spec.TiFlash.Replicas > 0 {
```

Comment:
Do we need to check the actual replica in sts status? Maybe sts is scaling in even if `spec.replicas` is 0.

Reply:
Deleting a scaling sts is still OK?

Reply:
seem ok

Reply:
https://github.com/pingcap/tidb-operator/pull/4842/commits/10594e24be3fc34671f9d8ea35bed265124c8c3d

Reply:
https://github.com/pingcap/tidb-operator/pull/4842/commits/0424db15ad4b2de0fb8c882525b7054dfcbe2387